## Before Raw Material Quantites

In [5]:
from scipy.optimize import linprog

# Coefficients for the objective function (maximize profit)
# Z = 2.70P - 0.40K - 0.45M - 0.25T
# Since linprog minimizes, we use the negative of the coefficients
c = [-2.70, 0.40, 0.45, 0.25]

# Coefficients for the inequality constraints (Ax <= b)
A = [
    [1, 1, 0, 0],  # Material X constraint: P + K <= 7500
    [2, 0, 1, 0],  # Material Y constraint: 2P + M <= 9000
    [-1, 1, 1, 1]  # Waste balance constraint: -P + K + M + T = 0
]

from scipy.optimize import linprog# Coefficients for the objective function (maximize profit)# Z = 2.70P - 0.40K - 0.45M - 0.25T# Since linprog minimizes, we use the negative of the coefficientsc = [-2.70, 0.40, 0.45, 0.25]# Coefficients for the inequality constraints (Ax <= b)A = [    [1, 1, 0, 0],  # Material X constraint: P + K <= 7500    [2, 0, 1, 0],  # Material Y constraint: 2P + M <= 9000    [-1, 1, 1, 1]  # Waste balance constraint: -P + K + M + T = 0]# Right-hand side of the inequality constraintsb = [7500, 9000, 0]# Bounds for the variables (P, K, M, T >= 0)bounds = [(0, None), (0, None), (0, None), (0, None)]# Solve the linear programming problemresult = linprog(c, A_ub=A, b_ub=b, bounds=bounds, method='highs')# Extract the resultsP, K, M, T = result.xmax_profit = -result.fun  # Convert back to maximization# Output the resultsprint(f"Optimal Production Plan:")print(f"Primary Product (P): {P:.2f} pounds")print(f"Secondary Product K (K): {K:.2f} pounds")print(f"Secondary Product M (M): {M:.2f} pounds")print(f"Liquid Waste Treated (T): {T:.2f} pounds")print(f"Maximum Profit: ${max_profit:.2f}")
# Right-hand side of the inequality constraints
b = [7500, 9000, 0]

# Bounds for the variables (P, K, M, T >= 0)
bounds = [(0, None), (0, None), (0, None), (0, None)]

# Solve the linear programming problem
result = linprog(c, A_ub=A, b_ub=b, bounds=bounds, method='highs')

# Extract the results
P, K, M, T = result.x
max_profit = -result.fun  # Convert back to maximization

# Output the results
print(f"Optimal Production Plan:")
print(f"Primary Product (P): {P:.2f} pounds")
print(f"Secondary Product K (K): {K:.2f} pounds")
print(f"Secondary Product M (M): {M:.2f} pounds")
print(f"Liquid Waste Treated (T): {T:.2f} pounds")
print(f"Maximum Profit: ${max_profit:.2f}")

Optimal Production Plan:
Primary Product (P): 4500.00 pounds
Secondary Product K (K): 0.00 pounds
Secondary Product M (M): 0.00 pounds
Liquid Waste Treated (T): 0.00 pounds
Maximum Profit: $12150.00


## After Raw Material Quantities

We will introduce two new decision variables to track the usage of raw materials:







In [4]:
from scipy.optimize import linprog

# Function to calculate and display raw material usage
def calculate_raw_material_usage(P, K, M):
    """
    Calculate the usage of raw materials X and Y based on production quantities.

    Args:
        P (float): Pounds of primary product produced.
        K (float): Pounds of secondary product K produced.
        M (float): Pounds of secondary product M produced.

    Returns:
        dict: A dictionary containing the usage of raw materials X and Y.
    """
    X_used = P + K  # Each pound of P and K requires 1 pound of X
    Y_used = 2 * P + M  # Each pound of P requires 2 pounds of Y, and each pound of M requires 1 pound of Y
    return {"X_used": X_used, "Y_used": Y_used}

# Coefficients for the objective function (maximize profit)
# Z = 2.70P - 0.40K - 0.45M - 0.25T
# Since linprog minimizes, we use the negative of the coefficients
c = [-2.70, 0.40, 0.45, 0.25]

# Coefficients for the inequality constraints (Ax <= b)
A = [
    [1, 1, 0, 0],  # Material X constraint: P + K <= 7500
    [2, 0, 1, 0],  # Material Y constraint: 2P + M <= 9000
    [-1, 1, 1, 1]  # Waste balance constraint: -P + K + M + T = 0
]

# Right-hand side of the inequality constraints
b = [7500, 9000, 0]

# Bounds for the variables (P, K, M, T >= 0)
bounds = [(0, None), (0, None), (0, None), (0, None)]

# Solve the linear programming problem
result = linprog(c, A_ub=A, b_ub=b, bounds=bounds, method='highs')

# Extract the results
P, K, M, T = result.x
max_profit = -result.fun  # Convert back to maximization

# Calculate raw material usage using the separate function
raw_material_usage = calculate_raw_material_usage(P, K, M)
X_used = raw_material_usage["X_used"]
Y_used = raw_material_usage["Y_used"]

# Output the results
print(f"Optimal Production Plan:")
print(f"Primary Product (P): {P:.2f} pounds")
print(f"Secondary Product K (K): {K:.2f} pounds")
print(f"Secondary Product M (M): {M:.2f} pounds")
print(f"Liquid Waste Treated (T): {T:.2f} pounds")
print(f"\nRaw Material Usage:")
print(f"Raw Material X Used: {X_used:.2f} pounds (Available: 7,500 pounds)")
print(f"Raw Material Y Used: {Y_used:.2f} pounds (Available: 9,000 pounds)")
print(f"\nMaximum Profit: ${max_profit:.2f}")

Optimal Production Plan:
Primary Product (P): 4500.00 pounds
Secondary Product K (K): 0.00 pounds
Secondary Product M (M): 0.00 pounds
Liquid Waste Treated (T): 0.00 pounds

Raw Material Usage:
Raw Material X Used: 4500.00 pounds (Available: 7,500 pounds)
Raw Material Y Used: 9000.00 pounds (Available: 9,000 pounds)

Maximum Profit: $12150.00


# ***Findings***

## Optimal Production Plan

Primary Product (P): The optimal production quantity is 4,500 pounds.

Secondary Product K (K): It is not profitable to produce secondary product K, so the optimal quantity is 0 pounds.

Secondary Product M (M): It is also not profitable to produce secondary product M, so the optimal quantity is 0 pounds.

Liquid Waste Treated (T): All liquid waste generated (4,500 pounds) should be specially treated and dumped into the river to comply with EPA regulations.

## Raw Material Usage

Raw Material X: The optimal plan uses 4,500 pounds of raw material X (out of 7,500 pounds available).
Raw Material Y: The optimal plan uses 9,000 pounds of raw material Y (fully utilizing the available 9,000 pounds).

## Maximum Profit

The optimal production and waste disposal plan yields a maximum profit of $12,150.

## Accountant's Recommendation

The accountant recommended eliminating product K as an alternative because it is not profitable. The analysis confirms this recommendation:
Producing product K incurs higher costs (raw material X and direct labor) than the revenue it generates.
Eliminating product K does not negatively impact the optimal solution.

## Sensitivity Analysis

The profit is highly sensitive to the availability of raw material Y, as it is fully utilized in the optimal plan. Acquiring additional raw material Y could increase production and profit.
The profit is less sensitive to raw material X, as there is unused capacity (3,000 pounds remaining).

## Waste Disposal Plan

The most cost-effective method for handling liquid waste is to specially treat it at a cost of $0.25 per pound. This ensures compliance with EPA regulations without incurring the higher costs associated with producing secondary products K or M.

# ***Recommendations***

## Production Strategy

Focus on producing the primary product (4,500 pounds) as it is the most profitable.
Do not produce secondary products K or M, as they are not profitable given the current costs and selling prices.

## Waste Disposal

Treat all liquid waste (4,500 pounds) to comply with EPA regulations. This is the most cost-effective option compared to converting waste into secondary products.

## Raw Material Management

Raw Material Y is a limiting factor in production. Acquiring additional raw material Y could increase production and profit.
Raw Material X has unused capacity (3,000 pounds), so no immediate action is needed unless production scales up.
Revisiting Costs and Prices:
If the costs of producing secondary products K or M decrease (e.g., lower labor costs or higher selling prices), reconsider their inclusion in the production plan.
Regularly review raw material costs and selling prices to ensure the production plan remains optimal.


# ***Conclusion***

The linear programming model provides a clear and optimal solution for SBIC to maximize profit while complying with EPA regulations. By producing 4,500 pounds of the primary product, treating all liquid waste, and avoiding the production of unprofitable secondary products, the company can achieve a maximum profit of $12,150. The analysis also highlights the importance of raw material Y as a limiting factor and validates the accountant's recommendation to eliminate product K. This approach ensures both profitability and environmental compliance.